shithead_game_server/src/lobby/mod.rs

```diff
@@ -63,45 +61,13 @@ pub enum LobbyState {
 /// A timer for the choosing 3 up lobby state.
 #[derive(Debug)]
 pub struct Choosing3UpState {
-    _timer: Choosing3UpTimer,
+    timeout_cancellation_token: CancellationToken,
     deck: CardsDeck,
 }
-
-/// A timer for the choosing 3 up lobby state.
-#[derive(Debug)]
-pub struct Choosing3UpTimer {
-    task: tokio::task::JoinHandle<()>,
-    notify: Arc<Notify>,
-}
-impl Choosing3UpTimer {
-    /// Creates a new timer.
-    pub fn new(lobby_id: LobbyId) -> Self {
-        let notify = Arc::new(Notify::new());
-        let notify_clone = Arc::clone(&notify);
-        Self {
-            task: tokio::spawn(async move {
-                match tokio::time::timeout(CHOOSE_TOP_3_DURATION, notify_clone.notified()).await {
-                    Ok(()) => {
-                        // we got a notification, which means that all client chose their 3 up
-                        // cards before the time was up, so we can just stop the timer.
-                        return;
-                    }
-                    Err(_) => {
-                        // if a timeout has occured, let the lobby know
-                        GAME_SERVER_STATE.choose_3_up_timeout(lobby_id);
-                    }
-                }
-            }),
-            notify,
-        }
-    }
-
-    pub async fn stop(self) {
-        // notifying this `Notify` object will notify the task because the task is waiting to
-        // receive a notification on it, and when the task will receive the notification it will
-        // stop.
-        self.notify.notify_one();
-        self.task.await.unwrap();
+impl Choosing3UpState{
+    /// Cancels the choosing 3 up timer.
+    pub fn cancel_timer(self){
```

Comment:
why this line is not async whilst `stop_next_turn_timer()` method inside the `Turn` struct actually is async. if it's required to be fixed please await it where required or don't make the `stop_next_turn_timer()` method async and remove the await